chatkit-core/src/main/kotlin/com/pusher/chatkit/CurrentUser.kt

```diff
@@ -37,47 +40,103 @@ class CurrentUser(
         get() = syncCurrentUser.rooms
 
 
+    /**
+     * Fetches all the users in the instance. The [callback] will be called with a [Result] when the
+     * operation is complete. If the operation was successful you will receive a [List<User>] which
+     * contains all the users, or an [Error] informing you of what went wrong.
+     */
     fun users(callback: (Result<List<User>, Error>) -> Unit) =
             makeCallback({ syncCurrentUser.users }, callback)
 
+    /**
+     * Returns whether the current user is subscribed to the [roomId]
+     */
     fun isSubscribedToRoom(roomId: String): Boolean =
             syncCurrentUser.isSubscribedToRoom(roomId)
 
+    /**
+     * @see [isSubscribedToRoom]
+     */
     fun isSubscribedToRoom(room: Room): Boolean =
             syncCurrentUser.isSubscribedToRoom(room)
 
+    /**
+     * Updates the current user with the properties of [newUser]
+     */
     fun updateWithPropertiesOf(newUser: CurrentUser) =
             syncCurrentUser.updateWithPropertiesOf(newUser.syncCurrentUser)
 
+    /**
+     * Sets the read cursor for the cur rent user in [room] to [position].
+     * The [position] is the id of the last message that the current user has read.
```

Comment:
Does this make sense `This will consequently notify any other connected users who have a [RoomListeners] set for [onNewReadCursor] to let them know the current user has read up to [position].` - i'm unsure if the current user is clear that it's yourself, and not the current user listening to events?

Reply:
Yes, that makes sense. Perhaps the second current user could be "this user" to help the ambiguity.

Setting your cursors also updates *your* unread count, both here and for all devices connected as your user. That's probably the key effect for most people calling this method.

Reply:
Ah i forgot about the unread count! Does this make sense:
```
* Sets the read cursor for the current user in [room] to [position].
     * The [position] is the id of the last message that the current user has read.
     * Setting your read cursor will update your unread count for this room for the current user.
     * This will additionally notify any other connected users who have a [RoomListeners]
     * set for [onNewReadCursor] to let them know this current user has read up to [position].
``` ?

Reply:
Yes, that's great